### ISSUE TITLE
Documentation not clear about the use of router.push with query params

### ISSUE THREAD

**Issue by ClementLevesque** [scottrippey/next-router-mock#122]
I struggled to understand how setup a test with a url and query params using router.push.

I started with the following code which does not set any query param :

```
await mockRouter.push('/page', {
  query: {
    param1: 'myParam',
  },
});
```

I then realized that this could lead to inconsistent params with query params in the url and the query object.

We need indeed to use the object syntax to make it work with the pathname :

```
await mockRouter.push({
  pathname: '/page',
  query: { param1: 'myParam' },
});
```

Maybe we could add an example in the docs ? I can propose something if you think it's worth it.

**Comment by scottrippey** [comment]
Is the first syntax supported by Next?  If they allow for `push(string, Partial<URL>)` then it's probably worthwhile to add support for that too.  But, I thought that it only supported `push(string) | push(URL)`. 

I'd be happy to add more examples in the docs, so feel free to open a PR with your suggestion! ❤️

**Comment by ClementLevesque** [comment]
> I thought that it only supported push(string) | push(URL).
You're right, it is a string or a Url Object, the second parameter is just a decorator for the URL shown in the browser url bar.
However I can maybe just write a short PR with an example using a URL Object

**Comment by ClementLevesque** [comment]
I finally think the the current examples are enough, there is no need to add more as this could lead to more confusion, thanks for your reply ;)
